src/schema/formats.yaml

```diff
@@ -0,0 +1,66 @@
+---
+# This file defines valid patterns for different formats
+# Entity patterns
+label:
+  description: |
+    Freeform labels without special characters.
+  pattern: ^[0-9a-zA-Z]+$
+index:
+  description: |
+    Non-negative, non-zero integers, optionally prefixed with leading zeros for sortability.
+  pattern: ^[0-9]+$
+# Metadata types
+string:
+  description: |
+    A basic string type (not a specific format).
+    This should allow any free-form string *except* "n/a".
+  pattern: ^(?!(n/a)$).*$
+integer:
+  description: |
+    An integer.
+  pattern: ^[+-]?\d+$
+number:
+  description: |
+    A number.
+  pattern: ^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$
+boolean:
+  description: |
+    A boolean.
+  pattern: ^(true|false)$
+# String patterns
+date:
+  description: |
+    A date.
+    YYYY-MM-DD[Z]
+  pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}?([A-Z]{3,4})$
+datetime:
+  description: |
+    A datetime.
+    YYYY-MM-DDThh:mm:ss[.000000][Z]
+  pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}?(\.[0-9]{1,6})?([A-Z]{3,4})$
+unit:
+  description: |
+    A unit.
+    Currently this matches any string that isn't "n/a"
+    TODO: Somehow reference the actual unit options in the Units appendix.
+  pattern: ^(?!(n/a)$).*$
+dataset_relative:
+  description: |
+    A path to a file, relative to the dataset folder.
+
+    The validation for this format is minimal.
+  pattern: ^[0-9a-zA-Z/_-\.]+$
+participant_relative:
+  description: |
+    A path to a file, relative to the participant's folder in the dataset.
+
+    The validation for this format is minimal.
+  pattern: ^[0-9a-zA-Z/_-\.]+$
+uri:
+  description: |
+    A uniform resource indicator.
+  pattern: ^[a-zA-Z]+:[0-9a-zA-Z/_-\.]+$
+bids_uri:
+  description: |
+    A BIDS uniform resource indicator.
+  pattern: ^bids:[0-9a-zA-Z/_-\.]+$
```

Comment:
The patterns for all four of the path/URI formats should be revised. I don't know enough about the rules to define solid patterns just yet.

Reply:
The same goes for `stimuli_relative` (added after the initial comment).